java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/DBEntityFromTables.java

```diff
@@ -0,0 +1,139 @@
+    "MSG_SelectTables=Select Database Tables",
+    "MSG_EnterPackageName=Enter package name"
+})
+@ServiceProvider(service = CodeActionsProvider.class)
+public class DBEntityFromTables extends CodeActionsProvider {
+
+    private static final String COMMAND_ENTITY_FROM_TABLES = "db.entity.from.tables"; //NOI18N
+
+    @Override
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (!COMMAND_ENTITY_FROM_TABLES.equals(command)) {
+            return null;
+        }
+        Lookup.getDefault().lookup(LspServerState.class).openedProjects().thenAccept((projects) -> {
+            if (projects.length > 0) {
+                createEntityClassesInProject(client, projects[0]);
+            } else {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoProject()));
+            }
+
+        });
+        return null;
+    }
+
+    private CompletableFuture createEntityClassesInProject(NbCodeLanguageClient client, Project prj) {
+        try {
+            DatabaseConnection connection = ConnectionManager.getDefault().getPreferredConnection(true);
+            if (connection == null) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoDbConn()));
+                return null;
+            }
+            ConnectionManager.getDefault().connect(connection);
+            SourceGroup[] sr = ProjectUtils.getSources(prj).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA);
+            if (sr.length < 1) {
+                client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoSourceRoot()));
+                return null;
+            }
+            Connection conn = connection.getJDBCConnection();
+            ResultSet rs = conn.getMetaData().getTables(conn.getCatalog(), conn.getSchema(), "%", new String[]{"TABLE", "VIEW"}); //NOI18N
+            List<QuickPickItem> dbItems = new ArrayList<>();
+            while (rs.next()) {
+                dbItems.add(new QuickPickItem(rs.getString("TABLE_NAME"))); //NOI18N
+            }
+            return client.showQuickPick(new ShowQuickPickParams(Bundle.MSG_SelectTables(), true, dbItems))
+                    .thenApply(items -> items.stream().map(item -> item.getLabel()).collect(Collectors.toList()))
+                    .thenAccept(tables -> {
+                        client.showInputBox(new ShowInputBoxParams(Bundle.MSG_EnterPackageName(), "")) //NOI18N
+                                .thenAccept(packageName -> {
+                                    EntitiesFromDBGenerator generator = new EntitiesFromDBGenerator(tables, true, packageName, sr[0], connection, prj, null);
+                                    ProgressContributor pc = BasicAggregateProgressFactory.createProgressContributor("entity"); //NOI18N
+                                    try {
+                                        generator.generate(pc);
```

Comment:
Also potentially run inside LSP event loop (input box response is sent as a protocol response to the server). Please check and if so, run asynchronously. The `ProgressContributor` indicates that the code called runs for some time and reports progress - that won't work when LSP connection is blocked.